### PR TITLE
refactor: fix mypy issues

### PR DESCRIPTION
**Pull request by madlabman**

### CODE REVIEW

File: src/providers/keys/client.py [hunk 1]
```diff
@@ -40,6 +40,6 @@ def _get_with_blockstamp(self, url: str, blockstamp: BlockStamp, params: Optiona
 
     @lru_cache(maxsize=1)
     @list_of_dataclasses(LidoKey.from_response)
-    def get_all_lido_keys(self, blockstamp: BlockStamp) -> list[LidoKey]:
+    def get_all_lido_keys(self, blockstamp: BlockStamp) -> list[dict]:
```
Comment (skhomuti) [inline src/providers/keys/client.py:43]: `list_of_dataclasses` affects the return type, the current variant is correct

File: src/providers/consensus/client.py [hunk 2]
```diff
@@ -67,15 +72,19 @@ def get_validators(self, blockstamp: BlockStamp) -> list[Validator]:
         return self.get_validators_no_cache(blockstamp)
 
     @list_of_dataclasses(Validator.from_response)
-    def get_validators_no_cache(self, blockstamp: BlockStamp, pub_keys: Optional[str | tuple] = None) -> list[Validator]:
+    def get_validators_no_cache(self, blockstamp: BlockStamp, pub_keys: Optional[str | tuple] = None) -> list[dict]:
```
Comment (skhomuti) [inline src/providers/consensus/client.py:75]: see `list_of_dataclasses`

File: src/web3py/contract_tweak.py [hunk 3]
```diff
@@ -1,20 +1,30 @@
 import itertools
```
Comment (skhomuti) [inline src/web3py/contract_tweak.py:1]: Probably it should be ignored because mostly it's just a copypaste from `web3py`

**Review by skhomuti — CHANGES_REQUESTED** [review]

File: src/modules/submodules/consensus.py [hunk 4]
```diff
@@ -330,7 +328,7 @@ def _get_slot_delay_before_data_submit(self, blockstamp: BlockStamp) -> int:
 
         members, _ = consensus_contract.functions.getMembers().call(block_identifier=blockstamp.block_hash)
 
-        mem_position = members.index(variables.ACCOUNT.address)
+        mem_position = members.index(variables.ACCOUNT.address)  # doesn't supposed to be used without private key provided?
```
Comment (F4ever) [inline src/modules/submodules/consensus.py:331]: Yeah, seems this is bug.
Definitely we should check if ACCOUNT is not None here.

File: src/providers/keys/client.py (hunk 1 above)
Comment (madlabman) [reply src/providers/keys/client.py:43]: `get_all_lido_keys` and similar functions return mappings, which is expected by the `list_of_dataclasses`. `list_of_dataclasses` is a separate function and its return type is inferred from its own typing.

File: src/web3py/contract_tweak.py (hunk 3 above)
Comment (madlabman) [reply src/web3py/contract_tweak.py:1]: Done

File: src/modules/submodules/consensus.py (hunk 4 above)
Comment (madlabman) [reply src/modules/submodules/consensus.py]: Done

**Review by F4ever — APPROVED** [review]

**Comment by F4ever** [comment]
Just fix mypy before merge

**Review by skhomuti — APPROVED** [review]

**Pull request merged**